src/services/role.service.ts

```diff
@@ -1,23 +1,31 @@
+import { GuildMember } from 'discord.js';
 import { RoleType, RoleTypeKey } from './../common/types';
-import { GuildMember, Role, Collection } from 'discord.js';
 
+// roles are a hierarchical enum/type. the following is the hierarchy (use these keys defind in src/common/types in 'RoleType'):
+//   Admin
+//   Moderator
+//   Teaching Assistant
+//   RegularUser
+//   Suspended
+// an Admin can run everythin a Mod and below can run, a Mod can run everythin a TA and below can run ...
+// exception:
+// Suspended role supersedes any other role.
+// i.e., a member with both the with the Suspended and Admin roles can only run what a Suspended user can run
 export class RoleService {
-  hasPermission(member: GuildMember, minRoleToRun: number) {
-    const roles = member.roles;
-    const highestRole = this._getHighestRole(roles.cache);
-    return highestRole >= minRoleToRun;
-  }
-
-  private _getHighestRole(roles: Collection<string, Role>) {
-    let highestRole = 0;
-
-    roles.forEach((role: Role) => {
-      if (role.name in RoleType) {
-        if (role.name === 'Suspended') {return RoleType.Suspended;}
-        highestRole = Math.max(highestRole, RoleType[<RoleTypeKey>role.name]);
+  hasPermission(member: GuildMember, minRoleToRun: RoleType): boolean {
+    let memberRole: RoleType = RoleType.RegularUser;
+    const orderedRoleTypes: RoleTypeKey[] = [
+      'Suspended',
+      'Admin',
+      'Moderator',
+      'Teaching Assistant',
+    ];
+    for (const roleType of orderedRoleTypes) {
+      if (member.roles.cache.find(role => role.name === roleType)) {
```

Comment:
```suggestion
      if (member.roles.cache.some(role => role.name === roleType)) {
```

Since we only care if an element passes a condition and not about the value of the element itself from `find`, `some` is more suitable here.

Reply:
Good point. I will change this.